File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size() ;
        unordered_map<int,int> comp(n) ;
        int a = -1 , b = -1 ;
        for(int i=0;i<n;i++){
            int c = target - nums[i] ;
            if(comp.find(c) != comp.end()){ a = i ; b = comp[c] ; break ;}
            else{
                comp[nums[i]]=i ;
            } 
        }

        vector<int> ans ;
        if(a<=b){
            ans.push_back(a);
            ans.push_back(b);
        }
        else{
            ans.push_back(b);
            ans.push_back(a);
        }

        return ans ;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:

        n = len(numbers) 
        my_map = dict() 


        for i in range(n):
            c = target- numbers[i] 
            if c in my_map:
                return [my_map[c]+1,i+1]
            
            if c not in my_map:
                my_map[numbers[i]] = i 
        

        return [0,0] ;